charts/actions-runner-controller/values.yaml

```diff
@@ -66,6 +66,7 @@ image:
 imagePullSecrets: []
 nameOverride: ""
 fullnameOverride: ""
+useRunnerStatusUpdateHookEphemeralRole: false
```

Comment:
I started to wonder if this can be improved after https://github.com/actions-runner-controller/actions-runner-controller/pull/1546.
One might want to choose if they want ARC to create a service account for the runner pod, and separately enable additional features like the kubernetes container mode and/or this status update hooks.
Considering all the possibles variations of configs, perhaps something like this would be better:

```
runner:
  serviceAccount:
    create: true # This will let ARC create a service account for each runner pod
    #name: <specify the name of existing SA. Makes sense only when `create: false`
  statusUpdateHook:
    enabled: true #=> This enables te runner status update hook and adds the role and role binding to either the existing or the dynamically created service account
  #Uncomment if you want the kubernetes container mode.
  # In case runner.serviceAccount.created=true,runner.statusUpdateHook.enabled=true,containerMode=kubernetes is specified,
  # ARC will create a runner service account for each runner pod and adds all the permissions required for both the container mode and the status update hooks
  #containerMode: kubernetes
```

Reply:
I believe these are indeed 2 different features but the scope a bit different. containerMode is something you want to add to your runners, while status update hook I believe is a more horizonal feature (e.g this PR adds a new column in the runner's table to show the running job, which applies for all runners. In other words I believe it makes more sense to be an on/off feature but to the controller - rather than be a switch per runner. While for the containerMode I do believe that makes sense in the context of the runner. Thoughts?

Reply:
Thanks for confirming! You're absolutely correct.

In my previous comment, I made two mistakes.

First, I confused the runner service account vs workflow job pod's service account. This feature affects the former and the new container mode is related to the latter.

Second, my suggested values.yaml snippet did look like containing settings for runners, not for the controller. Configuring runners is out of the scope of the ARC's chart. It might make sense a bit if it was for default runner configs, as we already have some runner defaults related values like `image.actionsRunnerRepositoryAndTag`. However, this feature should indeed be a controller-level feature as you pointed out.

Reply:
That said, perhaps it would look ok if we just omitted the containerMode key from my previous example?

```
runner:
  serviceAccount:
    create: true # This will let ARC create a service account for each runner pod
  statusUpdateHook:
    enabled: true #=> This enables the runner status update hook. If serviceaccount.create=true, this also creates the role and bind it to the dynamically created service account
```

To let ARC use an existing runner pod service account, you'd say:

```
runner:
  serviceAccount:
    name: mysa # The name of an existing SA. Valid only when `create: false`
    # In this case you don't need to give ARC the permission to create SA at all, which might be preferable to some folks for security reasons.
  statusUpdateHook:
    enabled: true # This doesn't automatically create a role and rolebinding associated to the existing SA, similarly to the top-level `serviceAccount.name` which his used to specify the existing SA for the controller-manager pod.
```

Perhaps it might be a good idea to imply `runner.serviceAccount.create=true` when you set `runner.statusUpdateHook.enabled=true`:

```
runner:
  statusUpdateHook:
    enabled: true #=> This enables the runner status update hook. In order to do so, ARC automatically creates a runner pod serviceaccount and a role with necessary permissions, and binds the role to the SA.
```

This way, it would look like a controller-wide on/off feature, while providing enough flexibility to let the user decide whether to bring their own SA for runner pods, or not.

WDYT?

Reply:
> First, I confused the runner service account vs workflow job pod's service account. This feature affects the former and the new container mode is related to the latter.

Sorry, this was again a complete misunderstanding of mine! The new container mode works by giving an existing SA to every runner pod so that it can create workflow job pods from within runner pods. Never mind about this. But I think my suggestion in https://github.com/actions-runner-controller/actions-runner-controller/pull/1268#discussion_r912317146 is still valid.
It's just that we won't need to provide the container mode related settings in helm charts as it's a runner-level feature.

Reply:
I think my suggestion in this thread can be distilled into just one sentence considering the scope of this pull request. That's "Can we rename useRunnerStatusUpdateHookEphemeralRole to runner.statusUpdateHook.enabled, so that the values will be kept structured great even if we added an ability to let ARC use a pre-created SA as the default runner pod SA in the future?"

In other words, even I mentioned it in my previous examples, we don't need to add support for `runner.serviceAccount.create` or `runner.serviceAccount.name` in this PR. That might be out of the scope of this PR.

Reply:
sounds good to me :) I've done the rename to `runner.statusUpdateHook.enabled` in [`ec02bb7` (#1268)](https://github.com/actions-runner-controller/actions-runner-controller/pull/1268/commits/ec02bb77a613d173d79364958a19b58c1e706b0b)